src/accountant_skel.rs

```diff
@@ -104,46 +111,87 @@ impl<W: Write + Send + 'static> AccountantSkel<W> {
         }
     }
 
+    fn verifier(
+        recvr: &streamer::PacketReceiver,
+        sendr: &Sender<(Vec<SharedPackets>, Vec<Vec<u8>>)>,
+    ) -> Result<()> {
+        let timer = Duration::new(1, 0);
+        let msgs = recvr.recv_timeout(timer)?;
+        trace!("got msgs");
+        let mut v = Vec::new();
+        v.push(msgs);
+        while let Ok(more) = recvr.try_recv() {
+            trace!("got more msgs");
+            v.push(more);
+        }
+        info!("batch {}", v.len());
+        let chunk = max(1, (v.len() + 3) / 4);
+        let chunks: Vec<_> = v.chunks(chunk).collect();
+        let rvs: Vec<_> = chunks
+            .into_par_iter()
+            .map(|x| gpu::ecdsa_verify(&x.to_vec()))
+            .collect();
+        for (v, r) in v.chunks(chunk).zip(rvs) {
+            sendr.send((v.to_vec(), r))?;
+        }
+        Ok(())
+    }
+
+    pub fn deserialize_packets(p: &packet::Packets) -> Vec<Option<(Request, SocketAddr)>> {
+        //deserealize in parallel
+        let mut r = vec![];
+        for x in &p.packets {
+            let rsp_addr = x.meta.addr();
+            let sz = x.meta.size;
+            if let Ok(req) = deserialize(&x.data[0..sz]) {
+                r.push(Some((req, rsp_addr)));
+            } else {
+                r.push(None);
+            }
+        }
+        r
```

Comment:
Where's the parallelization?

You can probably do something like this:
```rust
p.packets.par_iter().map(|x|  deserialize(&x.data[0..x.meta.size]).ok().map(|r| (r, x.meta.addr()))
```

But thinking you should leave that `Result` in the return value. That error message could be useful, especially for debugging. So:

```rust
p.packets.par_iter().map(|x|  deserialize(&x.data[0..x.meta.size]).map(|r| (r, x.meta.addr()))
```